### PR TITLE
scripts: Refactorgenerate_source.py

### PR DESCRIPTION
**Pull request by spencer-lunarg**
cc @mark-lunarg @mikes-lunarg the old `lvl_genvk.py` was in need of an update

- I renamed it `run_generator.py` and made it a module (to save calling `subprocess`)
- Removed all the arguments that seems to have been added but doubt ever used
- Removed `external_revision_generator` for a simpler `spirv_tool_commit_id_generator.py` file

### CODE REVIEW

**Review by mikes-lunarg — APPROVED** [review]
LGTM, just a couple comments

File: docs/generated_code.md [hunk 1]
```diff
@@ -40,7 +40,12 @@ NOTE: `VVL_CODEGEN` is `OFF` by default to allow users to build `VVL` via `add_s
 
 ## Tips
 
-If only dealing with a single file, comment out all the other file names in `scripts/generate_source.py` to speed up testing iterations.
+If only dealing with a single file,  run `scripts/generate_source.py` with `--target`
```
Comment (juan-lunarg) [inline docs/generated_code.md:43]: I like this CMake approach 💯

**Review by juan-lunarg — APPROVED** [review]
Cleanup commit messsage `Refactor generate_source.py`

File: scripts/generators/spirv_tool_commit_id_generator.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+from generators.generator_utils import (fileIsGeneratedWarning)
+from generators.base_generator import BaseGenerator
+
+class SpirvToolCommitIdOutputGenerator(BaseGenerator):
+    def __init__(self):
+        BaseGenerator.__init__(self)
+
+    def generate(self):
+        json_file = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'known_good.json'))
+        json_keys = 'repos,3,commit'
```
[the remark this answers is not in the document]
Comment (spencer-lunarg) [reply scripts/generators/spirv_tool_commit_id_generator.py:31]: ya.... lets 100% just do that lol

File: scripts/run_generator.py [hunk 3]
```diff
@@ -390,36 +414,37 @@ def makeGenOpts(args):
...
+    gen = createGenerator()
+
+    options = genOpts[target][1]
+
+    # Create the registry object with the specified generator and generator
+    # options. The options are set before XML loading as they may affect it.
+    reg = Registry(gen, options)
+
+    # Parse the specified registry XML into an ElementTree object
+    startTimer(time)
```
[the remark this answers is not in the document]
Comment (spencer-lunarg) [reply scripts/run_generator.py:431]: ok... me neither, I am happy to remove them as well

**Pull request merged**